gen.go

```diff
@@ -680,26 +715,31 @@ func emitCborMarshalArrayField(w io.Writer, f Field) error {
 
 func emitCborMarshalStructTuple(w io.Writer, gti *GenTypeInfo) error {
```

Comment:
Can we just bypass calling this entirely when "transparent" and just encode the single field directly? Maybe breaking the field code out into a new function?